ui/src/features/annotator/annotation-shape.tsx

```diff
@@ -14,19 +16,38 @@ const getFormattedPoints = (points: Point[]): string => {
 export const AnnotationShape = ({ annotation }: AnnotationShapeProps) => {
     const shape = annotation.shape;
     const color = annotation.labels[0].color;
+    const { setSelectedAnnotation, selectedAnnotation } = useAnnotator();
```

Comment:
We should try to avoid adding behaviour on top of the `AnnotationShape` as it's only purpose should be rendering its shape. This way we can independantly use it for our drawing tools as well as render it in the media grid.

Instead we should compose behaviour by adding a `<SelectableAnnotation >` component that we talked about here: https://github.com/open-edge-platform/training_extensions/pull/4621#issuecomment-3242052575

Reply:
Totally agree. I opened this PR to get feedback on this because once this tool is integrated, all others will follow the same approach, so i want to get this as right as possible